File: lib_arch/src/main/java/com/yyxnb/arch/base/Java8Observer.java
package com.yyxnb.arch.base;

import android.arch.lifecycle.DefaultLifecycleObserver;
import android.arch.lifecycle.LifecycleOwner;
import android.support.annotation.NonNull;
import android.util.Log;

import com.yyxnb.common.AppConfig;

import java.io.Serializable;

public class Java8Observer implements DefaultLifecycleObserver, Serializable {

    private String tag;

    public Java8Observer(String tag) {
        this.tag = tag;
    }

    @Override
    public void onCreate(@NonNull LifecycleOwner owner) {
        Log.e(tag, owner.getLifecycle().getCurrentState().name());
    }

    @Override
    public void onStart(@NonNull LifecycleOwner owner) {
        Log.w(tag, owner.getLifecycle().getCurrentState().name());
    }

    @Override
    public void onResume(@NonNull LifecycleOwner owner) {
        Log.w(tag, owner.getLifecycle().getCurrentState().name());
    }

    @Override
    public void onPause(@NonNull LifecycleOwner owner) {
        Log.w(tag, owner.getLifecycle().getCurrentState().name());
    }

    @Override
    public void onStop(@NonNull LifecycleOwner owner) {
        Log.w(tag, owner.getLifecycle().getCurrentState().name());
    }

    @Override
    public void onDestroy(@NonNull LifecycleOwner owner) {
        Log.e(tag, owner.getLifecycle().getCurrentState().name());
    }
}

File: lib_arch/src/main/java/com/yyxnb/arch/base/Java8Observer.kt
package com.yyxnb.arch.base

import android.util.Log
import androidx.lifecycle.LifecycleOwner
import com.yyxnb.common.interfaces.ILifecycle
import java.io.Serializable

/**
 * 打印页面生命周期
 *
 * @author yyx
 */
class Java8Observer(private val tag: String) : ILifecycle, Serializable {

    override fun onCreate(owner: LifecycleOwner) {
        Log.e(tag, owner.lifecycle.currentState.name)
    }

    override fun onStart(owner: LifecycleOwner) {
        Log.w(tag, owner.lifecycle.currentState.name)
    }

    override fun onResume(owner: LifecycleOwner) {
        Log.w(tag, owner.lifecycle.currentState.name)
    }

    override fun onPause(owner: LifecycleOwner) {
        Log.w(tag, owner.lifecycle.currentState.name)
    }

    override fun onStop(owner: LifecycleOwner) {
        Log.w(tag, owner.lifecycle.currentState.name)
    }

    override fun onDestroy(owner: LifecycleOwner) {
        Log.e(tag, owner.lifecycle.currentState.name)
    }

}